webapp/passport/js/MxN/protein.js

```diff
@@ -697,5 +739,171 @@ protein =
             $("#filterpeplength").text($("#rangesliderlength").slider("values", 0) +
                     " - " + $("#rangesliderlength").slider("values", 1));
         });
+    },
+
+     findMatchingHeavy: function(precursor, heavyPrecursors) {
+        let result;
+        heavyPrecursors.forEach(function(heavyPrecursor) {
+            if (heavyPrecursor.SampleFileId === precursor.SampleFileId) {
+                result = heavyPrecursor;
+            }
+        });
+        return result;
+    },
+
+    renderCVPlot: function(data, targetElement) {
+
+        const barAes = {
+            yLeft: 'cvMean',
+            x: 'cvType',
+            xSub: 'sequence',
+            color: 'cvType'
+        };
+
+        if (data.length) {
+
+            const cvBarLayer = new LABKEY.vis.Layer({
+                geom: new LABKEY.vis.Geom.BarPlot({
+                    clickFn: function (event, row) {
+                        for (let i = 0; i < protein.precursors.length; i++) {
+                            if (protein.precursors[i].AbbreviatedLabel === row.sequence) {
+                                protein.selectPrecursor(protein.precursors[i].PrecursorId, true);
+                            }
+                        }
+                    },
+                    hoverFn: function (row) {
+                        return row.cvType + ': ' + row.cvMean.toFixed(1) + '%\n' +
```

Comment:
encoding needed here?

Reply:
Thanks for checking. The docs are less definitive here so I tested by intentionally injecting '<script>alert("test");</script>'. The script didn't execute and I saw that string in the tooltip itself.